kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```diff
@@ -230,6 +231,15 @@ func (c *loadBalancerController) onUpdate(update bapi.Update) {
 func (c *loadBalancerController) acceptScheduledRequests(stopCh <-chan struct{}) {
 	log.Infof("Will run periodic IPAM sync every %s", timer)
 	t := time.NewTicker(timer)
+	for {
+		if err := c.ensureDatastoreUpgraded(); err != nil {
+			log.WithError(err).Error("Failed to upgrade load balancer's IPAM block affinities.  Unable to sync load balancer services.  Will retry.")
+			time.Sleep(10 * time.Second)
+			continue
+		}
+		break
+	}
+	log.Info("Successfully upgraded load balancer's IPAM block affinities.  Continuing to sync load balancer services.")
```

Comment:
Can we long this only when we actually make a change to any data?

Reply:
Library already logs completion; I'll remove this one.